src/hardware/serialport/softmodem.cpp

```diff
@@ -150,6 +150,16 @@ CSerialModem::CSerialModem(const uint8_t port_idx, CommandLine *cmd)
 		        GetPortNumber(), telnet_mode ? "enabled" : "disabled");
 	}
 
+	// Get the connect speed to report
+	if (getUintFromString("baudrate:", val, cmd)) {
+		val = val > 0 && val <= 57600 ? val : 57600;
+		LOG_MSG("SERIAL: Port %" PRIu8 " will report baud rate %d",
+			GetPortNumber(), val);
+	} else {
+		val = 57600;
+	}
+	snprintf(connect_string, 14, "CONNECT %d", val);
```

Comment:
Suggest asserting that `val` isn't bigger than the max.

Reply:
One of the dangers with snprintf is that the buffer and it's write size are disconnected: connect_string and 14; there's nothing making sure connect_string can actually hold 14 characters.

Staging has `safe_sprintf` that programmatically gets the size of fixes character buffers, so it's not up the programmer to keep these in-sync.

`safe_sprintf(connect_string, "CONNECT %d", val);`